wlauto/workloads/octaned8/__init__.py

```diff
@@ -0,0 +1,97 @@
+    assets folder within the dependencies folder.
+
+    Original source from::
+
+        https://github.com/v8/v8/wiki/D8%20on%20Android
+
+    """
+
+    parameters = [
+        Parameter('timeout', kind=int, default=120,
+                  description='Timeout, in seconds, for the script run time.'),
+    ]
+
+    supported_platforms = ['android']
+
+    executables = ['d8', 'natives_blob.bin', 'snapshot_blob.bin']
+
+    def setup(self, context):
+
+        for f in self.executables:
+            binFile = context.resolver.get(Executable(self, self.device.abi, f))
+            self.device_exe = self.device.install(binFile)
+
+        self.assets_path = os.path.join(self.dependencies_directory, 'assets')
+
+        if not os.path.exists(self.assets_path):
+            self.logger.debug('d8 assets not found.  Please add to dependencies folder')
+
+        self.device.execute('mkdir -p {}'.format(self.device.path.join(self.device.working_directory, 'assets')))
+
+        # Copy the test assests from the octane github project to a single assets folder on the device
+        for assetFile in os.listdir(self.assets_path):
+            assets_path = os.path.join(self.assets_path, assetFile)
+            on_device_file = self.device.path.join(self.device.working_directory, 'assets', assetFile)
+            self.device.push_file(assets_path, on_device_file)
+        for assetFile in os.listdir(os.path.join(self.assets_path, 'css')):
+            assets_path = os.path.join(self.assets_path, 'css', assetFile)
+            on_device_file = self.device.path.join(self.device.working_directory, 'assets', assetFile)
+            self.device.push_file(assets_path, on_device_file)
+        for assetFile in os.listdir(os.path.join(self.assets_path, 'js')):
+            assets_path = os.path.join(self.assets_path, 'js', assetFile)
+            on_device_file = self.device.path.join(self.device.working_directory, 'assets', assetFile)
+            self.device.push_file(assets_path, on_device_file)
+
+        device_file = self.device.path.join(self.device.working_directory, 'octaned8.output')
+        self.device.execute('rm -f {}'.format(device_file))
+
+    def run(self, context):
+        assets_dir = self.device.path.join(self.device.working_directory, 'assets')
+        device_file = self.device.path.join(self.device.working_directory, 'octaned8.output')
+        self.command = 'cd {}; {}/d8 ./run.js >> {} 2>&1'.format(assets_dir, self.device.binaries_directory, device_file)
+        self.output = self.device.execute(self.command, timeout=self.timeout, check_exit_code=False)
+
+    def update_result(self, context):
+        host_file = os.path.join(context.output_directory, 'octaned8.output')
+        device_file = self.device.path.join(self.device.working_directory, 'octaned8.output')
+        self.device.pull_file(device_file, host_file)
+        context.add_artifact('octaned8', host_file, 'data')
+
+    def teardown(self, context):
```

Comment:
If asset deployment is moved into `initialize()` (see comment above), asset removal will correspondingly need to be moved into `finalize()`